src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.TagSource.cs

```diff
@@ -169,25 +193,38 @@ private sealed partial class TagSource : ForegroundThreadAffinitizedObject
 
                 DebugRecordInitialStackTrace();
 
+                // Create the tagger-specific events that will cause the tagger to refresh.
                 _eventSource = CreateEventSource();
+                _onVisibilityChanged = () =>
+                {
+                    this.AssertIsForeground();
+
+                    // any time visibility changes, resume tagging on all taggers.  Any non-visible taggers will pause
+                    // themselves immediately afterwards.
+                    Resume();
```

Comment:
I admit it wasn't immediately obvious why this wasn't calling Pause if it was invisible, but I see the idea here is we only enter the paused state once we did have an event, which also doubles as the "we need to recompute once we regain focus" bit. Not sure if that deserves a comment here or perhaps on the paused flag.